### PR TITLE
Feature/api calls

### PR DESCRIPTION
**Pull request by EllieAzaveda**
### Is this a fix or a feature?
- Feature

### What is the change or fix?
- Added API calls file with fetches; Refactor to connect APIs; Attempt to fix "Back to Main Page" button - Still need to fix (throws an error)

### How should this be tested?
- In browser

### Linked Issues: 
- #6

### CODE REVIEW

File: src/Details/ShowDetails.js [hunk 1]
```diff
@@ -1,22 +1,42 @@
...
+        specificMovie: {},
+        movieSelected: movie.id || false,
+        error: ''
+      }
+  }
+
+  render() {
+    return (
+      <div className='details-card'>
+      <button onClick={() => this.handleClick(false)}>Home Page</button>
```
Comment (battan40) [inline src/Details/ShowDetails.js:19]: How are we passing handleClick through here?  Could we adjust the pathway to regain functionality?

**Review by battan40 — APPROVED** [review]

File: src/App/App.js [hunk 2]
```diff
@@ -1,37 +1,57 @@
...
         handleClick={this.handleClick}/>
       </article>
-      )
+    )
+  }
 
+  componentDidMount() {
+    fetchAllMovies()
+      .then(movieData => this.setState({ movies: movieData.movies }))
+      .catch(error => this.setState({ error: this.handleSpecificErr(error.message) }))
```
Comment (letakeane) [inline src/App/App.js:44]: Very nice job putting the error in state.

File: src/App/App.js [hunk 3]
```diff
@@ -1,37 +1,57 @@
...
   }
+
+  handleSpecificErr(err) {
```
Comment (letakeane) [inline src/App/App.js:47]: Love this!

File: src/Details/ShowDetails.js (hunk 1 above)
Comment (letakeane) [reply src/Details/ShowDetails.js:19]: Agreed - the naming of this function could be more semantic, as well

File: src/Movies/Movies.css [hunk 4]
```diff
@@ -11,7 +11,10 @@
   opacity: .9;
 }
 
+<<<<<<< HEAD
+=======
 
+>>>>>>> main
```
Comment (letakeane) [inline src/Movies/Movies.css:17]: Looks like we had a merge conflict that didn't get fully resolved here.

**Review by letakeane — COMMENTED** [review]
Nice work refactoring the app to use actual network requests! I suspect you probably ran into some async issues. The code looks fairly clean, and it seems like you've got a grasp of using function and class components 😄

**Pull request merged**